src/main/java/org/folio/list/rest/QueryClient.java

```diff
@@ -0,0 +1,33 @@
+package org.folio.list.rest;
+
+import org.folio.querytool.domain.dto.QueryDetails;
+import org.folio.querytool.domain.dto.QueryIdentifier;
+import org.folio.querytool.domain.dto.SubmitQuery;
+import org.springframework.cloud.openfeign.FeignClient;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+@FeignClient(name = "query")
+public interface QueryClient {
+
+  @PostMapping("")
+  QueryIdentifier executeQuery(@RequestBody SubmitQuery submitQuery);
+
+  @GetMapping("/{queryId}")
+  QueryDetails getQuery(@RequestHeader UUID queryId);
+
+  // New method to get sorted ids for query
```

Comment:
I'd remove these comments, since they only really make sense right up until the PR is merged